x-pack/plugins/apm/server/lib/transaction_groups/get_error_rate.ts

```diff
@@ -96,30 +99,36 @@ export async function getErrorRate({
 
   const noHits = resp.hits.total.value === 0;
 
-  const erroneousTransactionsRate =
-    resp.aggregations?.total_transactions.buckets.map((bucket) => {
-      const successful =
-        bucket[EVENT_OUTCOME].buckets.find(
-          (eventOutcomeBucket) =>
-            eventOutcomeBucket.key === EventOutcome.success
-        )?.count.value ?? 0;
+  if (!resp.aggregations) {
+    return { noHits, erroneousTransactionsRate: [], average: null };
+  }
 
-      const failed =
-        bucket[EVENT_OUTCOME].buckets.find(
-          (eventOutcomeBucket) =>
-            eventOutcomeBucket.key === EventOutcome.failure
-        )?.count.value ?? 0;
+  function calculateTransactionErrorPercentage(
+    outcomeResponse: AggregationResultOf<typeof outcomes, {}>
+  ) {
+    const successfulTransactions =
+      outcomeResponse.buckets.find(
+        (bucket) => bucket.key === EventOutcome.success
+      )?.count.value ?? 0;
+    const failedTransactions =
+      outcomeResponse.buckets.find(
+        (bucket) => bucket.key === EventOutcome.failure
+      )?.count.value ?? 0;
```

Comment:
Just codegolfing a bit on a Friday evening. WDYT about this? Feels less dense and more readable to me.

```suggestion
    const outcomes = Object.fromEntries(
      outcomeResponse.buckets.map(({ key, count }) => [key, count.value ?? 0])
    );
    const failedTransactions = outcomes[EventOutcome.failure];
    const successfulTransactions = outcomes[EventOutcome.success];
```